apis/v1alpha2/gateway_types.go

```diff
@@ -518,12 +518,19 @@ const (
 	// Support: Extended
 	IPAddressType AddressType = "IPAddress"
 
-	// An opaque identifier that represents a specific IP address. The
-	// interpretation of the name is dependent on the controller. For
-	// example, a "NamedAddress" might be a cloud-dependent identifier
-	// for a static or elastic IP.
+	// A hostname represents a DNS based ingress point. This is similar to the
+	// corresponding hostname field in Kubernetes load balancer status. For
+	// example, this concept may be used for cloud load balancers where a DNS
+	// name is used to expose a load balancer.
 	//
-	// Support: Implementation-specific
+	// Support: Extended
+	HostnameAddressType AddressType = "Hostname"
+
+	// A NamedAddress provides a way to reference a specific IP address by name.
+	// For example, this may be a name or other unique identifier that refers
+	// to a static IP address on a cloud provider.
```

Comment:
```suggestion
	// A NamedAddress provides a way to reference a network endpoint by name.
	// For example, this may be a unique identifier that refers to a a cloud
	// provider-specific resource.
```

Getting completely rid of IP here might get the creativity of users flowing better.

Reply:
Update to make this a bit more generic, but wanted to leave some reference to a static IP as an example so there was at least one concrete use case referenced by the spec.